### PR TITLE
#167765385 Swagger API documentation setup

### PR DESCRIPTION
**Pull request by voke1**
### What does this PR do?

- Setup swagger API documentation

### Description of Task to be completed?

- Install and import `swagger-ui-express`
- Add swaggerDoc.json file to root directory.

### How should this be manually tested?

- Open `http://localhost:3000/api-docs/` in browser

### Any background context you want to provide?

-`Request` and `User` endpoint is set in the documentation for sample purposes

### What are the relevant pivotal tracker stories?

- #167765385

### Implementation Plan
- https://drive.google.com/file/d/1rox9mhsOGVyZDt4NKuR8tAT2TtJTgrUk/view?usp=sharing

Screenshot-api-documention
screenshot-ap-documentation-2

### CODE REVIEW

**Review by debelistic — APPROVED** [review]
Great job @voke1, well done

**Comment by voke1** [comment]
Thanks @debelistic

**Review by uggoprince — APPROVED** [review]
@voke1 it's well configured :+1:

did you use yml files or code comments?

**Comment by voke1** [comment]
> @voke1 it's well configured 👍
> 
> did you use yml files or code comments?

I used a json file. Swagger gives the option to use either yaml or json file. You will need to convert yaml file to json format.

File: src/index.js [hunk 1]
```diff
@@ -10,11 +10,17 @@ const fs = require("fs"),
     errorhandler = require("errorhandler"),
     mongoose = require("mongoose");
 
+import {serve, setup} from 'swagger-ui-express';
+import swaggerDoc from '../swaggerDoc';
```
[the remark this answers is not in the document]
Comment (anneKay) [reply src/index.js]: Please update this PR with develop.

[the remark this answers is not in the document]
Comment (voke1) [reply src/index.js]: > Please update this PR with develop.

Done

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+WORKDIR /usr/app
+
+# copy package.json
+COPY package.json .
+
+# install node modules (--quiet to specify)
+RUN npm install --quiet
+
+# copy all the app's files from the project's root to our working directory
+COPY . .
```
Comment (anneKay) [inline Dockerfile]: Add newline here

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+      - PORT
+  # set up postgres service to be used as dependecy
+  postgres:
+    # use predefined postgres image (-alpine)
+    image: postgres:9.6.2-alpine
+    environment:
+      # create user postgres as the default user for postgreSQL
+      - POSTGRES_USER
+      # create the default database as todos
+      - POSTGRES_DB
```
Comment (anneKay) [inline docker-compose.yml]: Add newline here

File: swaggerDoc.json [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+            "xml": {
+                "name": "User"
+            }
+        },
+        "externalDocs": {
+            "description": "Find out more on WeTravel site",
+            "url": "https://github.com/andela/phoenix-bn-backend"
+        }
+    }
+}
```
Comment (anneKay) [inline swaggerDoc.json]: Add new line here

Comment (voke1) [reply swaggerDoc.json]: Done

File: docker-compose.yml (hunk 3 above)
Comment (voke1) [reply docker-compose.yml]: Done

File: Dockerfile (hunk 2 above)
Comment (voke1) [reply Dockerfile]: Done

**Comment by anneKay** [comment]
Good job @voke1, did you follow the routes we already in the code for this documentation. I noticed you have an `api/vi/request` and `api/vi/auth/signin` Do we have such paths in the codebase?

**Comment by voke1** [comment]
HI @anneKay. Not yet, we don't have any path in the codebase already. The two routes included are just for test purposes. They can be deleted or modified when we commence with documenting of endpoints. do you advise I remove them?

**Comment by anneKay** [comment]
Please look in routes/api/users.js` file to see the routes we have

**Comment by voke1** [comment]
Oh Sorry, I didn't follow the routes we have already in the codebase. Do you mean I document these routes also? It was concluded in previous meeting that anyone working on a route should also add documentation for the route. Please advise.

**Comment by anneKay** [comment]
In that case we can leave it as it is, let's consider the routes you added test routes for setting up the documentation

**Pull request merged**